include/vast/Dialect/HighLevel/HighLevelTypes.hpp

```diff
@@ -157,8 +159,15 @@ namespace vast::hl
         }
     }
 
-    mlir::FunctionType getFunctionType(PointerType functionPointer);
-    mlir::FunctionType getFunctionType(mlir::Type functionPointer);
+    mlir::FunctionType getFunctionType(Type function_pointer, Module mod);
+
+    mlir::FunctionType getFunctionType(Value callee);
+    mlir::FunctionType getFunctionType(mlir::CallOpInterface call);
+    mlir::FunctionType getFunctionType(mlir::CallInterfaceCallable callee, Module mod);
+    mlir::FunctionType getFunctionType(mlir::CallInterfaceCallable callee, Module mod);
+
+    Type getTypedefType(TypedefType type, Module mod);
+    Type getBottomTypedefType(TypedefType def, Module mod);
```

Comment:
What does `Bottom` mean here? The first type that is not typedef anymore? Add doc.

Reply:
Yes it unwraps all typedefs, do you have a better name?

Reply:
Nope, I guess just add a comment in case someone is confused when reading it?